RocketChat/Rocket.Chat
# [IMPROVE] Removes hardcoded prune message limit of 2000

- [x] I have read the Contributing Guide - https://github.com/RocketChat/Rocket.Chat/blob/develop/.github/CONTRIBUTING.md#contributing-to-rocketchat
- I have added tests that prove my fix is effective or that my feature works (if applicable)
- I have added necessary documentation (if applicable)
- Any dependent changes have been merged and published in downstream modules

## Proposed changes (including videos or screenshots)


Added a new setting in the `General` section of the administrator panel. This setting allows the user to select how many messages the `Prune Messages` action will delete in bulk, this enables the user to dynamically set a limit compatible with their server's capability.

![image](https://user-images.githubusercontent.com/23219524/112572714-7ffa8700-8dc9-11eb-870f-6b228ece0b9f.png)




## Issue(s)

Closes #21305
## Steps to test or reproduce


1. Click the user Menu
2. Go to the admin panel
3. Go to the `General` section
4. Scroll down until you see the `Prune Messages Limit` input
5. Set the limit of your choice
6. Go to a channel
7. Click the channel menu
8. Select `Prune Messages`
9. Prune all channel messages

Now you should see a success message saying how many messages were deleted
![image](https://user-images.githubusercontent.com/23219524/112573642-7a05a580-8dcb-11eb-89a3-f94e47916010.png)


## Further comments
